### PR TITLE
Fix auth0 signup

### PR DESCRIPTION
**Pull request by oetherington**
Auth0 have an open bug that makes it impossible to connect to a Postgres database over SSL in custom DB scripts due to SSL throwing "self signed certificate in certificate chain". This breaks several of our integrations, but most importantly it makes it impossible to signup with email (OAuth providers seem to be unaffected).

After speaking with Auth0 support, they suggested upgrading node-pg and setting       `process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0'` as a temporary workaround, which does indeed seem to work. I'll keep in touch with support though so we can remove this hack ASAP.

I've also included a few small changes to fix some code-rot.

These changes have already been deployed to prod on Auth0, so no further action is needed after merging.

### CODE REVIEW

**Review by jpaddison3 — APPROVED** [review]

File: scripts/db/src/login.ts [hunk 1]
```diff
@@ -130,6 +130,13 @@ async function login(
         port: POSTGRES_PORT ? parseInt(POSTGRES_PORT) : 5432,
         ssl: TEMPLATE_DATA.pgShouldSsl,
       }
+
+      /**
+       * NOTE: Temporary fix for Auth0 bug August 2022
+       * Should be reverted ASAP
+       */
+      process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0'
```
Comment (s-cheng) [inline scripts/db/src/login.ts:138]: @oetherington Do we need to do this for logins? I thought logging in with username/pw worked fine? I was at least able to do it yesterday.

**Review by s-cheng — COMMENTED** [review]
@oetherington I chatted with my roommate who has some relevant expertise here. He said that after this code is removed, we should update our postgres password. It's possible for it to be leaked and we might as well account for that.

He also suggested that the issue might be that Auth0 no longer trusts our CA, so we can try fixing this by passing it in directly in `ssl`, as seen [here](https://node-postgres.com/features/ssl).

File: scripts/db/src/login.ts (hunk 1 above)
Comment (oetherington) [reply scripts/db/src/login.ts:138]: Unfortunately, it seems yes. We only make a postgres request if the user isn't found in the main forum database though, so this is why it worked for you yesterday.

**Comment by oetherington** [comment]
@s-cheng Yeah - we tried `ssl.rejectUnauthorized` yesterday but it didn't fix the issue for some reason. The fact that it works by setting the environment variable but not by setting the parameter directly doesn't make much sense to me, but this is a pretty wacky bug anyway so it's probably not worth trying to reason about too deeply right now unless we can get some more info from Auth0.

I agree it's probably wise to change the PG password after this - I don't think it should be strictly necessary, but it can't hurt.

**Pull request merged**